src/werkzeug/_internal.py

```diff
@@ -161,12 +178,12 @@ def _has_level_handler(logger):
         if not current.propagate:
             break
 
-        current = current.parent
+        current = current.parent  # type: ignore
 
     return False
 
 
-def _log(type, message, *args, **kwargs):
+def _log(type: str, message: str, *args, **kwargs) -> None:
```

Comment:
And also the incomplete-defs rule, but I haven't enable either of those so I didn't go out of my way to use `Any` everywhere.

Reply:
This would be better as `*args: object, **kwargs: object`

Reply:
That doesn't give any benefit right now, and `object` is unhelpfully restrictive compared to `Any` in most contexts.

Reply:
`object` is completely unrestricted

Reply:
I think you’re confused. See if https://stackoverflow.com/questions/39817081/typing-any-vs-object helps.

Reply:
https://mypy.readthedocs.io/en/stable/dynamic_typing.html#any-vs-object

> only operations valid for all types are accepted for object values

While it *accepts* all values, it then restricts what you can do on them. So for example if those kwargs are passed through to another function that does type them, there will need to be unecessary casts all over the place. This entire discussion is moot, as typing/mypy isn't really expressive enough to describe heterogeneous splat args.

Rather than trying to continue that discussion in this sub thread on a merged PR, your effort would probably be better spent on some of the further work I outlined in the initial comment of this PR. ;-)